File: Python/simple_factory/pizza_factory.py
from pizza import CheesePizza, PepperoniPizza, Pizza


class SimplePizzaFactory:

    def create_pizza(self, type_: str) -> Pizza:
        pizza: Pizza = None
        if type_ == "cheese":
            pizza = CheesePizza()
        elif type_ == "pepperoni":
            pizza = PepperoniPizza()
        return pizza

File: Golang/simple_factory/pizza_factory.go
package simple_factory

import "fmt"

type pizzaFactory interface {
	CreatePizza(pizzaType string) (IPizza, error)
}

type SimplePizzaFactory struct {}

func (spf *SimplePizzaFactory) CreatePizza(pizzaType string) (IPizza, error) {
	switch pizzaType {
	case "cheese":
		return NewCheesePizza(), nil
	case "pepperoni":
		return NewPepperoniPizza(), nil
	}

	return nil, fmt.Errorf("invalid pizza type")

}